app/src/main/java/com/github/se/polyfit/model/ingredient/Ingredient.kt

```diff
@@ -0,0 +1,74 @@
+package com.github.se.polyfit.model.ingredient
+
+import com.github.se.polyfit.model.meal.Meal
+import kotlin.properties.Delegates
+
+/**
+ * Represents an ingredient in a meal. Automatically updates the parent meal when any property
+ * changes.
+ *
+ * @property name is unique for each meal
+ */
+class Ingredient(
+    val name: String, // name is now a val
+    servingSize: Double,
+    calories: Double,
+    protein: Double,
+    carbohydrates: Double,
+    fat: Double
+) {
+  var servingSize: Double by
+      Delegates.observable(servingSize) { _, _, _ -> parentMeal?.updateMeal() }
+  var calories: Double by Delegates.observable(calories) { _, _, _ -> parentMeal?.updateMeal() }
+  var protein: Double by Delegates.observable(protein) { _, _, _ -> parentMeal?.updateMeal() }
+  var carbohydrates: Double by
+      Delegates.observable(carbohydrates) { _, _, _ -> parentMeal?.updateMeal() }
+  var fat: Double by Delegates.observable(fat) { _, _, _ -> parentMeal?.updateMeal() }
+
+  var parentMeal: Meal? = null
```

Comment:
Then ingredient will belong to serving instead of a meal, and it can belong to many servints not just one